src/monkey-patch.android.ts

```diff
@@ -1,39 +1,40 @@
-// @ts-expect-error because there is no corresponding type definition
 import * as NativeAndroidManager from "react-native/Libraries/Components/StatusBar/NativeStatusBarManagerAndroid";
+const RCTStatusBarManagerCompat = require("./specs/NativeStatusBarManagerCompat").default;
 
-const DefaultNativeAndroidManager = NativeAndroidManager.default;
-const getConstants = NativeAndroidManager.default.getConstants;
-const RCTStatusBarManagerCompat =
-  require("./specs/NativeStatusBarManagerCompat").default;
+// Copy original default manager to keep its original state and methods
+const OriginalNativeAndroidManager = {...NativeAndroidManager.default};
 
+// Create a new object that modifies the necessary methods
 // On Android < 11 RN uses legacy API which breaks EdgeToEdge mode in RN, so
 // in order to use library on all available platforms we have to monkey patch
 // default RN implementation and use modern `WindowInsetsControllerCompat`.
-export const applyMonkeyPatch = () => {
-  NativeAndroidManager.default = {
-    getConstants,
-    setColor(color: number, animated: boolean): void {
-      RCTStatusBarManagerCompat.setColor(color, animated);
-    },
-
-    setTranslucent(translucent: boolean): void {
-      RCTStatusBarManagerCompat.setTranslucent(translucent);
-    },
-
-    /**
+const ModifiedNativeAndroidManager = {
+  ...NativeAndroidManager.default, // Spread original properties to keep existing functionality
+  setColor: (color: number, animated: boolean): void => {
+    RCTStatusBarManagerCompat.setColor(color, animated);
+  },
+  setTranslucent: (translucent: boolean): void => {
+    RCTStatusBarManagerCompat.setTranslucent(translucent);
+  },
+   /**
      *  - statusBarStyles can be:
      *    - 'default'
      *    - 'dark-content'
      */
-    setStyle(statusBarStyle?: string): void {
-      RCTStatusBarManagerCompat.setStyle(statusBarStyle);
-    },
+  setStyle: (statusBarStyle?: string): void => {
+    RCTStatusBarManagerCompat.setStyle(statusBarStyle);
+  },
+  setHidden: (hidden: boolean): void => {
+    RCTStatusBarManagerCompat.setHidden(hidden);
+  },
+};
 
-    setHidden(hidden: boolean): void {
-      RCTStatusBarManagerCompat.setHidden(hidden);
-    },
-  };
+// Define a function to apply the monkey patch
+export const applyMonkeyPatch = () => {
+  Object.assign(NativeAndroidManager.default, ModifiedNativeAndroidManager);
```

Comment:
Thanks! I'll check how it works today and if everything is okay, then I'll merge it 👍

Reply:
had a little help from gpt!